### PR TITLE
New MVC for webservices component

### PR DESCRIPTION
**Pull request by fastslack**
This is an initial approach to new MVC move. All controllers seems to be working fine but test on ajax is needed. I changed the old way to call ajax to a new way using an specific controller (AjaxController.php). Save option seems to be working fine, creating the xml file and wsdl too, but also needs a detailed test.

All comments are very welcome.
## Test instructions

1) Get GIT files
2) Install Joomla!
3) link or copy $GIT/component/admin to $WWWROOT/administrator/components/com_webservices 
4) copy $GIT/component/webservices.xml to $WWWROOT/administrator/components/com_webservices  (should we copy it to $GIT/component/admin?)
5) Go to discover on Joomla! installation and install the extension
6) Edit $WWWROOT/administrator/components/com_webservices/webservices.php line 17 and put the real path of the GIT files
7) Edit $GIT/config.json and set your database credentials

### CODE REVIEW

File: src/Renderer/Hal.php [hunk 1]
```diff
@@ -105,9 +105,9 @@ public function render($cache = false, $params = array())
 		$this->app->setHeader('Expires', '0', true);
 		$this->app->setHeader('Cache-Control', 'must-revalidate, post-check=0, pre-check=0', true);
 		$this->app->setHeader('Cache-Control', 'private', false);
-		$this->app->setHeader('Content-Type', $this->getMimeEncoding() . '; charset=' . $this->getCharset(), true);
-		$this->app->setHeader('Webservice-Name', $this->webservice->webserviceName, true);
-		$this->app->setHeader('Webservice-Version', $this->webservice->webserviceVersion, true);
+		$this->app->setHeader('Content-type', $this->getMimeEncoding() . '; charset=' . $this->getCharset(), true);
+		$this->app->setHeader('Webservice name', $this->webservice->webserviceName, true);
+		$this->app->setHeader('Webservice version', $this->webservice->webserviceVersion, true);
```
Comment (chrisdavenport) [inline src/Renderer/Hal.php]: Can you update your code?  You are reverting some fixes made a while back.  Thanks.

Comment (fastslack) [inline src/Renderer/Hal.php]: Ups. Done.

**Comment by chrisdavenport** [comment]
Thanks @fastslack 

Can you update the component/webservices.xml file to reflect the changes made to the file structure?

File: component/admin/Webservices/Table/WebserviceTable.php [hunk 2]
```diff
@@ -1,35 +1,23 @@
...
 use Joomla\Database\DatabaseDriver;
 
 /**
  * Webservice table.
  *
- * @package     Redcore.Backend
- * @subpackage  Tables
- * @since       1.4
- *
- * @property  int     $id
```
Comment (wilsonge) [inline component/admin/Webservices/Table/WebserviceTable.php:21]: Can we readd the property tags please?

**Comment by wilsonge** [comment]
At a quick glance over this looks really impressive! I'll make sure I have a play around during the week :)

**Comment by javigomez** [comment]
Super! Thanks Matias

**Comment by javigomez** [comment]
Agg Im in mobile and cant do proper inline comments. Ill check in deep later but just found by chance a small typo at component/admin/Webservices/Helper.php. line 14,should say Webservices Component instead.

**Comment by fastslack** [comment]
Merged tests changes. Thanks @javigomez

**Comment by javigomez** [comment]
Tests are still Work in progress. Still a few changes are needed to make it finally work. but almost there.

**Pull request closed without merging**